protostar/starknet/execution_environment.py

```diff
@@ -21,15 +21,15 @@ def __init__(self, state: ExecutionState):
         self.state: ExecutionState = state
 
     @abstractmethod
-    async def invoke(self, function_name: str) -> InvokeResultT:
+    async def execute(self, function_name: str) -> InvokeResultT:
         ...
 
-    async def perform_invoke(
+    async def perform_execute(
         self, function_name: str, *args, **kwargs
     ) -> StarknetCallInfo:
         try:
             func = getattr(self.state.contract, function_name)
-            return await func(*args, **kwargs).invoke()
+            return await func(*args, **kwargs).execute()
```

Comment:
Is `execute` a new name for a method of StarkNet functions or did you rename this method accidentally?

Reply:
Its the new name:

https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/testing/contract.py#L281-L293